artemis-server/src/main/java/org/apache/activemq/artemis/core/server/ServerSession.java

```diff
@@ -111,7 +112,7 @@ public interface ServerSession extends SecurityAuth {
 
    void addCloseable(Closeable closeable);
 
-   boolean checkAutoCreate(SimpleString address, RoutingType routingType) throws Exception;
+   AutoCreateResult checkAutoCreate(QueueConfiguration queueConfiguration) throws Exception;
```

Comment:
I like this idea, but I don't want it to be part of this PR as it has already grown quite a bit larger than I originally anticipated. Also, care will be necessary to limit the static methods so as to avoid the same kind of "constructor hell" that I originally implemented `QueueConfiguration` to address (among other things).

Reply:
if you aren't going to do that then I'd suggest going through and cleaning up the if statements where you are creating objects and configuring them with internal boolean logic for routing type etc as it create rather unreadable logic statements.

Reply:
I'll try to simplify it.